File: CursoEmVideo/POO [60 Horas]/PHP POO [40 Horas]/Aula 06/ControleRemoto.php
<?php
    require_once 'Controlador.php';
    class ControleRemoto implements Controlador {
        private $volume;
        private $ligado;
        private $tocando;

        function __construct() {
            $this->setVolume(50);
            $this->setLigado(true);
            $this->setTocando(false);
        }

        public function getVolume() {
            return $this->volume;
        }
    
        private function setVolume($volume) {
            $this->volume = $volume;
        }
    
        private function getLigado() {
            return $this->ligado;
        }
    
        private function setLigado($ligado) {
            $this->ligado = $ligado;
        }
    
        private function getTocando() {
            return $this->tocando;
        }
    
        private function setTocando($tocando) {
            $this->tocando = $tocando;
        }

        public function ligar() {
            $this->setLigado(true);
        }
    
        public function desligar() {
            $this->setLigado(false);
        }
    
        public function abrirMenu() {
            print("----- MENU -----\n");
            print("Está ligado? ".$this->getLigado()."\n");
            print("Está tocando? ".$this->getTocando()."\n");
            print("Volume: ".$this->getVolume()."\n");
            for ($i = 0; $i < $this->getVolume(); $i += 10) {
                print("| ");
            }
            print("\n");
        }
    
        public function fecharMenu() {
            print("Fechando menu\n");
        }
    
        public function maisVolume() {
            if ($this->getLigado()) {
                $this->setVolume($this->getVolume() + 5);
            } else {
                print("Impossível aumentar volume.\n");
            }
        }
    
        public function menosVolume() {
            if ($this->getLigado()) {
                $this->setVolume($this->getVolume() - 5);
            } else {
                print("Impossível diminuir volume.\n");
            }
        }
    
        public function ligarMudo() {
            if (($this->getLigado()) && ($this->getVolume() > 0)) {
                $this->setVolume(0);
            }
        }
    
        public function desligarMudo() {
            if (($this->getLigado()) && ($this->getVolume() == 0)) {
                $this->setVolume(50);
            }
        }
    
        public function play() {
            if (($this->getLigado()) and !($this->getTocando())) {
                $this->setTocando(true);
            } else {
                print("Não consegui reproduzir\n");
            }
        }
    
        public function pause() {
            if (($this->getLigado()) && ($this->getTocando())) {
                $this->setTocando(false);
            } else {
                print("Não consegui pausar\n");
            }
        }
    }

File: CursoEmVideo/POO [60 Horas]/Java POO [40 Horas]/Aula 06/ControleRemoto.java
public class ControleRemoto implements Controlador {
    private int volume;
    private boolean ligado;
    private boolean tocando;

    public ControleRemoto() {
        setVolume(50);
        setLigado(true);
        setTocando(false);
    }

    private int getVolume() {
        return volume;
    }

    private void setVolume(int volume) {
        this.volume = volume;
    }

    private boolean isLigado() {
        return ligado;
    }

    private void setLigado(boolean ligado) {
        this.ligado = ligado;
    }

    private boolean isTocando() {
        return tocando;
    }

    private void setTocando(boolean tocando) {
        this.tocando = tocando;
    }

    @Override
    public void ligar() {
        this.setLigado(true);
    }

    @Override
    public void desligar() {
        this.setLigado(false);
    }

    @Override
    public void abrirMenu() {
        System.out.println("----- MENU -----");
        System.out.println("Está ligado? " + this.isLigado());
        System.out.println("Está tocando? " + this.isTocando());
        System.out.print("Volume: " + this.getVolume());
        for (int i = 0; i < this.getVolume(); i += 10) {
            System.out.print("| ");
        }
        System.out.println();
    }

    @Override
    public void fecharMenu() {
        System.out.println("Fechando menu");
    }

    @Override
    public void maisVolume() {
        if (this.isLigado()) {
            this.setVolume(this.getVolume() + 5);
        } else {
            System.out.println("Impossível aumentar volume.");
        }
    }

    @Override
    public void menosVolume() {
        if (this.isLigado()) {
            this.setVolume(this.getVolume() - 5);
        } else {
            System.out.println("Impossível diminuir volume.");
        }
    }

    @Override
    public void ligarMudo() {
        if ((this.isLigado()) && (this.getVolume() > 0)) {
            this.setVolume(0);
        }
    }

    @Override
    public void desligarMudo() {
        if ((this.isLigado()) && (this.getVolume() == 0)) {
            this.setVolume(50);
        }
    }

    @Override
    public void play() {
        if ((this.isLigado()) && (!this.isTocando())) {
            this.setTocando(true);
        } else {
            System.out.println("Não consegui reproduzir");
        }
    }

    @Override
    public void pause() {
        if ((this.isLigado()) && (this.isTocando())) {
            this.setTocando(false);
        } else {
            System.out.println("Não consegui pausar");
        }
    }

}
